allennlp/data/dataset_readers/huggingface_datasets_reader.py

```
@@ -0,0 +1,190 @@
+
+            # datasets Value can be of different types
+            elif isinstance(value, Value):
+
+                # String value maps to TextField
+                if value.dtype == "string":
+                    # Since TextField has to be made of Tokens add whole text as a token
+                    # TODO Should we use simple heuristics to identify what is token and what is not?
+                    field = TextField([Token(inputs[0][feature])])
+
+                else:
+                    field = LabelField(
+                        inputs[0][feature], label_namespace=feature, skip_indexing=True
+                    )
+
+            elif isinstance(value, Sequence):
+                # datasets Sequence of strings to ListField of TextField
+                if value.feature.dtype == "string":
+                    field_list = list()
+                    for item in inputs[0][feature]:
+                        item_field = TextField([Token(item)])
+                        field_list.append(item_field)
+
+                    if len(field_list) == 0:
+                        continue
+                    field = ListField(field_list)
+
+                # datasets Sequence of strings to ListField of LabelField
+                elif isinstance(value.feature, ClassLabel):
+                    field_list = list()
+                    for item in inputs[0][feature]:
+                        item_field = LabelField(
+                            label=item, label_namespace=feature, skip_indexing=True
+                        )
+                        field_list.append(item_field)
+                    if len(field_list) == 0:
+                        continue
+                    field = ListField(field_list)
+
+            # datasets.Translation cannot be mapped directly
+            # but it's dict structure can be mapped to a ListField of 2 ListField
+            elif isinstance(value, Translation):
+                if value.dtype == "dict":
```

What happens if it's not a `dict`?

---

Documentation for _datasets_ indicates that it will be dict
https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Translation

---

Can you throw an exception (`ValueError` maybe?) otherwise then, with an explanation? Otherwise we'll see some sort of `NotInitializedException` later, which would be very confusing for a user.

---

Done.

---

Done, throwing ValueError indicating `feature` is not supported. Added this as else for similar if/elif elsewhere